beacon-chain/state/stateV0/state_trie.go

```diff
@@ -31,7 +32,40 @@ var (
 
 // InitializeFromProto the beacon state from a protobuf representation.
 func InitializeFromProto(st *pbp2p.BeaconState) (*BeaconState, error) {
-	return InitializeFromProtoUnsafe(proto.Clone(st).(*pbp2p.BeaconState))
+	clonedState, err := CloneBeaconState(st)
+	if err != nil {
+		return nil, err
+	}
+	return InitializeFromProtoUnsafe(clonedState)
+}
+
+// CloneBeaconState serves as a wrapper for proto.Clone to apply any
+// post-cloning changes needed. This was added due to a change in protobuf v2
+// which encodes empty slices as nil.
+func CloneBeaconState(st *pbp2p.BeaconState) (*pbp2p.BeaconState, error) {
```

Comment:
Is this sitll necessary? I thought that SszDeepEqual fixed this issue?

Reply:
Testing would fail with `nil validators in state`. Maybe its a problem that can be solved without this change. Let me see.